setup.py

```diff
@@ -152,7 +166,15 @@ def setup_package():
 
         metadata['version'] = VERSION
     else:
-        from numpy.distutils.core import setup
+        if is_numpy_installed:
+            from numpy.distutils.core import setup
+        else:
+            try:
+                from setuptools import setup
+            except ImportError:
+                sys.stderr.write("Numpy and setuptools not found\n"
+                                 "scikit-learn requires either of these to continue\n")
```

Comment:
I would not talk about setuptools in this error message, just numpy. numpy is required in anycase. setuptools is not mandatory, just helpful to get numpy installed automatically when missing.